controllers/kyma_controller.go

```diff
@@ -202,23 +203,25 @@ func (r *KymaReconciler) syncModuleCatalog(ctx context.Context, syncContext *rem
 	return nil
 }
 
-func (r *KymaReconciler) stateHandling(ctx context.Context, kyma *v1alpha1.Kyma) (ctrl.Result, error) {
+func (r *KymaReconciler) stateHandling(ctx context.Context, kyma *v1alpha1.Kyma,
+	syncContext *remote.KymaSynchronizationContext,
+) (ctrl.Result, error) {
 	log.FromContext(ctx).Info("syncing state", "state", string(kyma.Status.State))
 	switch kyma.Status.State {
 	case "":
 		return ctrl.Result{}, r.HandleInitialState(ctx, kyma)
 	case v1alpha1.StateProcessing:
-		return ctrl.Result{Requeue: true}, r.HandleProcessingState(ctx, kyma)
+		return ctrl.Result{Requeue: true}, r.HandleProcessingState(ctx, kyma, syncContext)
```

Comment:
The `ctx` is set during controller start, so that is not a good option, even if use unique kyma keys as keys for `context.WithValue(...)`

Reply:
Just because you use ctx during controller start does not mean the values set during the control-loop propagate upstream or cross boundaries.

If you call `context.WithValue(ctx, key, value)` in `Reconcile`, it will only pass the object in the context used within one reconciliation, thus it is safe to use even with a single key, there will be no references after Reconcile is over, and it will also be GCed normally.